File: onlinevote/src/main/java/com/example/vote/onlinevote/service/MyUserDetailsService.java
package com.example.vote.onlinevote.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

import com.example.vote.onlinevote.model.Login;
import com.example.vote.onlinevote.model.LoginPrincipal;
import com.example.vote.onlinevote.repository.LoginRepository;

@Service
public class MyUserDetailsService implements UserDetailsService {

    // This service will load user-specific data for authentication and authorization
    @Autowired
    private LoginRepository loginRepository;

    @Override
    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {

        
        Login login = loginRepository.findByUsername(username);

        if (login == null) {
            throw new UsernameNotFoundException("User not found");
        }

        return new LoginPrincipal(login);

        
    }

}

File: Kotlin/login/src/main/kotlin/com/relirou/login/service/MyUserDetailsService.kt
package com.relirou.login.service

import org.springframework.stereotype.Service
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UsernameNotFoundException
import com.relirou.login.repository.UserRepository
import com.relirou.login.model.UserPrincipal

@Service
class MyUserDetailsService (
    private val userRepository : UserRepository
) : UserDetailsService {

    override fun loadUserByUsername(username: String): UserDetails { 

        val users = userRepository.findByName(username)

        if(users == null) {
            println("User not found")
            throw UsernameNotFoundException("User not found")
        }

        return UserPrincipal(users);
    }
}